---
title: "Hi-C-LSTM: Learning representations of chromatin contacts using a recurrent neural network identifies genomic drivers of conformation"
collection: phd
status: revision
permalink: /phd/hiclstm
excerpt: 'Despite the availability of chromatin conformation capture experiments, discerning the relationship between the 1D genome and 3D conformation remains a challenge, which limits our understanding of their affect on gene expression and disease. We propose Hi-C-LSTM, a method that produces low-dimensional latent representations that summarize intra-chromosomal Hi-C contacts via a recurrent long short-term memory (LSTM) neural network model. We find that these representations contain all the information needed to recreate the observed Hi-C matrix with high accuracy, outperforming existing methods. These representations enable the identification of a variety of conformation-defining genomic elements, including nuclear compartments and conformation-related transcription factors. They furthermore enable in-silico perturbation experiments that measure the influence of cis-regulatory elements on conformation.'
date: 2021-10-01
venue: 'Nature Communications'
paperurl: 'https://assets.researchsquare.com/files/rs-878825/v1_covered.pdf?c=1633106140'
citation: 'Dsouza, K., Maslova, A., Al-Jibury, E., Merkenschlager, M., Bhargava, V., & Libbrecht, M. (2021). Hi-C-LSTM: Learning representations of chromatin contacts using a recurrent neural network identifies genomic drivers of 3D genome conformation.'
---